orangecontrib/text/corpus.py

```diff
@@ -225,12 +230,23 @@ def from_file(cls, filename):
         table = Table.from_file(filename)
         return cls(table.X, table.Y, table.metas, table.domain, None)
 
+    def ngrams_iterator(self, join_with=None):
+        if join_with is None:
+            return (list(chain(*(nltk.ngrams(doc, n)
+                    for n in range(self.ngram_range[0], self.ngram_range[1]+1))))
+                    for doc in self.tokens)
+        else:
+            return (list(chain(*((join_with.join(ngram) for ngram in nltk.ngrams(doc, n))
+                    for n in range(self.ngram_range[0], self.ngram_range[1] + 1))))
+                    for doc in self.tokens)
```

Comment:
@nikicc here is my suggestion for the ngrams issue. 
This method allows iterate through the lists of ngrams.